### PR TITLE
Builder IsInfinite and IsNaN

### PR DESCRIPTION
**Pull request by trenouf**
Added builder methods for IsInfinite and IsNaN, and switched the SPIR-V
reader to use them.

Change-Id: I530a3b20ed6186543fe0316dd6365c91cd29d41d

### CODE REVIEW

**Comment by trenouf** [comment]
Please review just the topmost commit.

File: builder/llpcBuilderImplArith.cpp [hunk 1]
```diff
@@ -972,6 +972,16 @@ Value* BuilderImplArith::FDivFast(
...
 }
 
+// =====================================================================================================================
+// Create "isInfinite" operation: return true if the supplied FP (or vector) value is infinity
+Value* BuilderImplArith::CreateIsInfinite(
+    Value*        pInValue,   // [in] Input value
+    const Twine&  instName)   // [in] Name to give instruction(s)
+{
+    // 0x004: negative infinity; 0x200: positive infinity
+    return CreateCallAmdgcnClass(pInValue, 0x204, instName);
```
Comment (linqun) [inline builder/llpcBuilderImplArith.cpp]: please add a bit-field struct or enums to represent the bits of amdgcn.class in llpcIntrinsDefs.h

File: builder/llpcBuilderRecorder.cpp [hunk 2]
```diff
@@ -121,6 +121,10 @@ StringRef BuilderRecorder::GetCallName(
         return "derivative";
     case Opcode::FMed3:
         return "fmed3";
+    case Opcode::IsInfinite:
+        return "is.infinite";
+    case Opcode::IsNaN:
+        return "is.nan";
```
Comment (amdrexu) [inline builder/llpcBuilderRecorder.cpp]: The same as above

**Comment by trenouf** [comment]
V2: Rebase
V3: Rebase on FMF changes.
V4: Addressed review comments.

**Comment by amdvlk-admin** [comment]
### Test summary for commit 2ec79abf664a83958d6e1173ab27c5e0fdcef928
<details open><summary>Driver commits used in build</summary><ul>

<li>CWPACK: amd-master b601c88aeca7a7b08becb3d32709de383c8ee428</li>
<li>METROHASH: amd-master 2b6fee002db6cc92345b02aeee963ebaaf4c0e2f</li>
<li>PAL: dev 88d997710b4e405f3a8e3fd60a38afee9e3e77e2</li>
<li>SPVGEN: dev 2f31d1170e8a12a66168b23235638c4bbc43ecdc</li>
<li>XGL: dev 2cb5558b94c5dc839e093cb439057a1802426c8e</li>
<li>LLVM: amd-vulkan-dev dcb42575c3d374467bb6e4e609f8bf6d1302c297</li>
</ul></details>

<details open><summary>ShaderDB test</summary><ul>

<li>Passed: 927/927 (100.0%)</li>
<li>Failed: 0/927 (0.0%)</li>
</ul></details>

<details open><summary>CTS tests (built with branch Vulkan-CTS-1.1.2.3-cherry-picks)</summary><ul>


<details open><summary>Ubuntu 18.04, Ellesmere</summary><ul>

<li>Passed: 68331/68331 (100.0%)</li>
<li>Failed: 0/68331 (0.0%)</li>
</ul></details>

<details open><summary>Rhel 7.6, Vega10</summary><ul>

<li>Passed: 68363/68363 (100.0%)</li>
<li>Failed: 0/68363 (0.0%)</li>
</ul></details>


</ul></details>

**Pull request merged**